### PR TITLE
Port to 1.21

### PR DESCRIPTION
**Pull request by Gaming32**
This also removes the `bingo:item_broken` trigger in favor of `minecraft:item_durability_changed`.

### CODE REVIEW

File: common/src/main/java/io/github/gaming32/bingo/util/ResourceLocations.java [hunk 1]
```diff
@@ -4,14 +4,16 @@
...
 
 public final class ResourceLocations {
+    private static final ResourceLocation BINGO_PROTOTYPE = ResourceLocation.fromNamespaceAndPath(Bingo.MOD_ID, "");
+
     private ResourceLocations() {
     }
 
     public static ResourceLocation bingo(String path) {
-        return new ResourceLocation(Bingo.MOD_ID, path);
+        return BINGO_PROTOTYPE.withPath(path);
```
Comment (Earthcomputer) [inline common/src/main/java/io/github/gaming32/bingo/util/ResourceLocations.java:13]: Hmm, why not just use `ResourceLocation.withNamespaceAndPath` directly?

File: common/src/main/generated/.cache/0863d6b6d415911b15e9b245a98fa603d174e7c6 [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
-e69e73d172c40891d65fb28f3dbbc50a77c89c3c data/bingo/tags/items/fishing_treasure.json
-c05e998c718feb7c79e577ca9f09edc7225b3d22 data/bingo/tags/items/flowers.json
-84ca487e773b4a6e5d2dfb9c9a4e6a7d99f6f480 data/bingo/tags/items/glazed_terracotta.json
-fadcbb38e7665bed6ea2c1b8876f4954189179c5 data/bingo/tags/items/gold_in_name.json
-1d06a2ee122e3edcd98600fe98e131c26392d4bf data/bingo/tags/items/living_coral_blocks.json
-76ff0180f4ac33ecd2e129331abfc90bb84adad2 data/bingo/tags/items/meat.json
-085879b3966fa2b81ee5d552455c5a238c0419ae data/bingo/tags/items/not_meat.json
-002cb871d0959e8ffd4cd0a7c52e8bed9800b392 data/bingo/tags/items/slabs.json
-79abb83176b057e09e9904aca8bcef5879d64bb2 data/bingo/tags/items/stairs.json
+// 1.21 Release Candidate 1
```
Comment (Earthcomputer) [inline common/src/main/generated/.cache/0863d6b6d415911b15e9b245a98fa603d174e7c6:1]: Datagen needs to be rerun

**Review by Earthcomputer — CHANGES_REQUESTED** [review]

File: common/src/main/java/io/github/gaming32/bingo/util/ResourceLocations.java (hunk 1 above)
Comment (xpple) [reply common/src/main/java/io/github/gaming32/bingo/util/ResourceLocations.java:13]: Or just partially apply the function `ResourceLocation.fromNamespaceAndPath` if you really want that prototype.

Comment (Gaming32) [reply common/src/main/java/io/github/gaming32/bingo/util/ResourceLocations.java:13]: I'm doing this to cache the validity of the namespace. Every time you run `fromNamespaceAndPath`, it rechecks if the namespace is valid. `fromPath` avoids this check.

Comment (Earthcomputer) [reply common/src/main/java/io/github/gaming32/bingo/util/ResourceLocations.java:13]: Should probably add a comment then because that's not clear

**Review by Earthcomputer — APPROVED** [review]

**Pull request merged**